pkg/metrics/producers/scheduledcapacity/producer.go

```diff
@@ -15,15 +15,76 @@ limitations under the License.
 package scheduledcapacity
 
 import (
+	"fmt"
 	"github.com/awslabs/karpenter/pkg/apis/autoscaling/v1alpha1"
+	"github.com/awslabs/karpenter/pkg/metrics"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/ptr"
+	"time"
 )
 
 // Producer implements the ScheduledCapacity metric
 type Producer struct {
 	*v1alpha1.MetricsProducer
+	ResourceName string
 }
 
 // Reconcile of the metrics
 func (p *Producer) Reconcile() error {
+	var pastRecommendation, futureRecommendation int32 = 0, 0
+	pastTime, futureTime := &apis.VolatileTime{Inner: metav1.Time{}}, &apis.VolatileTime{Inner: metav1.Time{}}
+
+	timeRanges := p.Spec.ScheduledCapacity.Schedules
+
+	for _, timeRange := range timeRanges {
+		start, err := time.Parse(time.RFC3339, timeRange.StartTime)
```

Comment:
I haven't dug into it, but make sure you're using the right thing:
https://stackoverflow.com/questions/522251/whats-the-difference-between-iso-8601-and-rfc-3339-date-formats

It's possible this is already correct. Might be worth a comment about ISO8601 and RFC3339 here and which interface we support.